File: Projetos/Alura/Guaruja/app/src/main/java/alura/com/guaruja/MainActivity.java
package alura.com.guaruja;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        WebView webNavegador = (WebView) findViewById(R.id.webview);
        webNavegador.loadUrl("https://guaruja.ossbrasil.com");
        webNavegador.setWebViewClient(new WebViewClient());
        WebSettings webSettings = webNavegador.getSettings();
        webSettings.setJavaScriptEnabled(true);}


        // Use When the user clicks a link from a web page in your WebView
        private class MyWebViewClient extends WebViewClient {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                if (Uri.parse(url).getHost().equals("http://meulink.com/")) {
                    return false;
                }
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
                startActivity(intent);
                return true;
            }
        }
    }

File: Projetos/Lista de games/app/src/main/java/vinibza/com/listadegames/MainActivity.kt
package vinibza.com.listadegames

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*
import vinibza.com.listadegames.model.AdapterGame
import vinibza.com.listadegames.model.Game

class MainActivity : AppCompatActivity() {

    lateinit var gameAdapter: AdapterGame

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initRecycler()
        initDados()
    }


    private fun initRecycler() {
        this.gameAdapter = AdapterGame({
                Game -> openLink( Game.link )
        })

        recycler_principal.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = gameAdapter
        }
    }


    private fun openLink(url: String) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        startActivity(intent)

    }


    private fun initDados(){
        val listaGames = DataSource.setDataSet()
        gameAdapter.setDados(listaGames)
    }

}